Tests

dkryvi opened the pull request.

dkryvi committed:
Prepare api for testing

dkryvi committed:
Write unit and integration tests

akosenko commented on app/utils/initializer.js:
```diff
@@ -1,10 +1,28 @@
+  elasticSearchInitialization(),
+])
+  .then(() => {
+    logger.info('Successfully initialize all services');
```
All services initialized successfully

dkryvi replied:
Ok. I'll fix

dkryvi replied:
[Done](https://github.com/inst-event-emitter/eventemitter-backend/pull/13/commits/7e10bfd671ee954c3b1eef19d05ccb6b0e9c31a4#diff-96a8661bff3c7b9cefe5405b208b2830R24)

dkryvi committed:
Update test and travis scripts

dkryvi committed:
Create docker-compose for test env. Setup elastic in test env

dkryvi committed:
Update message when all services are initialized
Define event factory for tests
Update test and dev elasticsearch configs
Create events in elasticsearch instead mock search method

Vanish404 commented on README.md:
```diff
@@ -1,4 +1,28 @@
+`docker-compose -f docker-compose.test.yml up`
+
+This command creates elastic search container.
+You ay connect to Elasticsearch through 127.0.0.1:9200 or 127.0.0.1:9300.
```
I think you made a mistake at this sentence
`ay`

dkryvi replied:
[Fixed](https://github.com/inst-event-emitter/eventemitter-backend/pull/13/commits/03b617748894f02a8f8d8baa3f39a267c570be3d#diff-04c6e90faac2675aa89e2176d2eec7d8R12)

Vanish404 commented on app/utils/initializer.js:
```diff
@@ -1,10 +1,28 @@
+    if (err) {
+      logger.error(`elasticsearch connection error: ${err}`);
+      reject(err);
+    } else {
```
in my opinion, i should say that we can refuse `else` statement, you can `return` one of callback functions. i'm talking about `resolve` and `reject`

dkryvi replied:
Yeap. I [update](https://github.com/inst-event-emitter/eventemitter-backend/pull/13/commits/03b617748894f02a8f8d8baa3f39a267c570be3d#diff-96a8661bff3c7b9cefe5405b208b2830R14) this code

Vanish404 commented on app/utils/initializer.js:
```diff
@@ -1,10 +1,28 @@
+const { elasticSearchClient } = require('../services/elasticSearch');
+
+const elasticSearchInitialization = () => new Promise((resolve, reject) => {
+  elasticSearchClient.ping({
```
I think, you need to `return` this method

dkryvi replied:
Why?

Vanish404 replied:
I read documentation about `Promise`, i'm sorry, i was wrong

dkryvi committed:
Fix minor bugs

dkryvi committed:
Locally resolve conflicts

Validate date when create new event (#14)

* Validate date when create new event

* Update validation schema name
Send all validation messages

Add ability to filter events by date range (#15)

dkryvi committed:
Merge branch 'dev' into tests
